### DISCUSSION TITLE
Public Processors for common scenarios?

### DISCUSSION THREAD

**Discussion by sunildatla** [open-telemetry/opentelemetry-dotnet#4852] [General]
Do we have any processors publicly within opentelemetry for dotnet community for logs which we can re-use if we need to for common scenarios like for example 

1.Throttling of logs 

2.PII/Secret masking 

3.LogRecord Size reduction (If the size of a log reaches above certain limit to truncate the log attributes)

**Comment by cijothomas** [comment]
1. None exists to my knowledge. I am not fully sure what is meant by "throttling of logs"? If you are worried that too many logs are being produced, you can use[ Filtering to reduce them](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/logs/customizing-the-sdk#log-filtering).
2. Check this guide: https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/logs/redaction It shows a possible way, you'll still have to write the processor yourself as there are no such processors available in this repo!
3. Same as above. Also, this is something that can be achieved at Exporter level. For eg: if you use OTLP Log Exporter, exporter offers mechanism to truncate attributes. If you use [GenevaExporter ](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Exporter.Geneva)from contrib repo, it also has some truncation mechanisms.

**Reply by sunildatla** [reply]
Thanks @cijothomas 
1. Filtering is used to remove categories all together for a particular tracelevel , it doesnt support us to throttle logs after certain rate per category for example . 
2. Thanks , will look into it 
3. We are using GenevaExporter , can you show the truncation mechanisms in geneva exporter . Is it truncating attributes all together or truncating the string to limit the size of each attribute. How can we use truncation mechanisms of genevaexporter ?

**Reply by cijothomas** [reply]
_it doesnt support us to throttle logs after certain rate per category for example_

The Filtering is part of ILogger itself, not an OpenTelemetry feature. If ILogger's filtering is not sufficient, its a good feature request to ILogger. OpenTelemetry further allows filtering with its processor pipeline. No public ones available, so you'll need to write yourself based on actual throttling logic.

**Reply by cijothomas** [reply]
> GenevaExporter 

It is not a component maintained in this repo. Please use the right support option for questions on that componet.
